### PR TITLE
deprecation(log): rename `warning()` to `warn()`

### PR DESCRIPTION
**Pull request by timreichen**
ref: https://github.com/denoland/deno_std/issues/4099

### CODE REVIEW

File: log/test.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ Deno.test("defaultHandlers", async function () {
   } = {
     DEBUG: log.debug,
     INFO: log.info,
-    WARNING: log.warning,
+    WARNING: log.warn,
```
Comment (joehillen) [inline log/test.ts:26]: `WARNING` should be changed to `WARN`

File: log/logger.ts [hunk 2]
```diff
@@ -179,11 +179,21 @@ export class Logger {
     return this.#_log(LogLevels.INFO, msg, ...args);
   }
 
+  /** @deprecated (will be removed after 0.214.0) Use {@linkcode warn} instead. */
   warning<T>(msg: () => T, ...args: unknown[]): T | undefined;
   warning<T>(msg: T extends GenericFunction ? never : T, ...args: unknown[]): T;
```
Comment (iuioiua) [inline log/logger.ts:184]:
```suggestion
  /** @deprecated (will be removed after 0.214.0) Use {@linkcode warn} instead. */
  warning<T>(msg: T extends GenericFunction ? never : T, ...args: unknown[]): T;
```
So both overloads show as deprecated.

File: log/logger.ts [hunk 3]
```diff
@@ -179,11 +179,21 @@ export class Logger {
...
+    return this.warn(msg as () => T, ...args);
+  }
+
+  warn<T>(msg: () => T, ...args: unknown[]): T | undefined;
+  warn<T>(msg: T extends GenericFunction ? never : T, ...args: unknown[]): T;
+  warn<T>(
+    msg: (T extends GenericFunction ? never : T) | (() => T),
+    ...args: unknown[]
   ): T | undefined {
     return this.#_log(LogLevels.WARNING, msg, ...args);
```
Comment (iuioiua) [inline log/logger.ts:198]:
```suggestion
    return this.#_log(LogLevels.WARN, msg, ...args);
```

File: log/mod.ts [hunk 4]
```diff
@@ -498,7 +498,11 @@ export function info<T>(
...
 }
 
-/** Log with warning level, using default logger. */
+/**
+ * @deprecated (will be removed after 0.214.0) Use {@linkcode warn} instead.
+ *
+ * Log with warning level, using default logger.
+ */
 export function warning<T>(msg: () => T, ...args: unknown[]): T | undefined;
 export function warning<T>(
```
Comment (iuioiua) [inline log/mod.ts:507]:
```suggestion
/**
 * @deprecated (will be removed after 0.214.0) Use {@linkcode warn} instead.
 *
 * Log with warning level, using default logger.
 */
export function warning<T>(
```

File: log/mod.ts [hunk 5]
```diff
@@ -510,12 +514,30 @@ export function warning<T>(
...
+  if (msg instanceof Function) {
+    return getLogger("default").warn(msg, ...args);
   }
-  return getLogger("default").warning(msg, ...args);
+  return getLogger("default").warn(msg, ...args);
 }
 
-/** Log with error level, using default logger. */
+/** Log with e
+ * rror level, using default logger. */
```
Comment (iuioiua) [inline log/mod.ts:540]: Please revert

File: log/test.ts (hunk 1 above)
Comment (iuioiua) [reply log/test.ts:26]: I agree

**Review by iuioiua — CHANGES_REQUESTED** [review]
Nice to see this done! Few nits.

File: log/test.ts (hunk 1 above)
Comment (timreichen) [reply log/test.ts]: done

File: log/mod.ts (hunk 5 above)
Comment (timreichen) [reply log/mod.ts]: done

**Review by iuioiua — APPROVED** [review]
LGTM! Nice.

Am I correct in understanding that this change is non-breaking?

**Comment by kt3k** [comment]
> Am I correct in understanding that this change is non-breaking?

Strictly speaking, if the users do any log analysis using `WARNING` symbol, then that analysis will break. Can we delay the change of output string from `WARNING` to `WARN` to the actual breaking version? (`after 0.214.0`)

**Comment by timreichen** [comment]
> > Am I correct in understanding that this change is non-breaking?
> 
> Strictly speaking, if the users do any log analysis using `WARNING` symbol, then that analysis will break. Can we delay the change of output string from `WARNING` to `WARN` to the actual breaking version? (`after 0.214.0`)

Reverted

**Comment by kt3k** [comment]
@timreichen I don't see well what part is reverted?

The below still seems printing `WARN hi`
```js
import * as log from "./log/mod.ts";
log.warning("hi");
```

**Comment by timreichen** [comment]
> @timreichen I don't see well what part is reverted?
> 
> The below still seems printing `WARN hi`
> 
> ```js
> import * as log from "./log/mod.ts";
> log.warning("hi");
> ```

Sorry, seems like the commit wasn't pushed. Done now.

**Review by kt3k — APPROVED** [review]
LGTM

**Review by iuioiua — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**